lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -990,9 +993,15 @@ newtype SlotNo = SlotNo { unSlotNo :: Word32 }
     deriving stock (Show, Read, Eq, Ord, Generic)
     deriving newtype (Num, Buildable, NFData, Enum)
 
-newtype EpochNo = EpochNo { unEpochNo :: Word64 }
+newtype EpochNo = EpochNo { unEpochNo :: Word31 }
     deriving stock (Show, Read, Eq, Ord, Generic)
-    deriving newtype (Num, Buildable, NFData, Enum)
+    deriving newtype (Num, Bounded, Enum)
```

Comment:
So this is the source of your question about `NFData` :) 
Looking more closely at the source code, `OddWord` is actually a `newtype` constructor around a `Word32`. Which promises no particular runtime overhead of the type itself. Meaning that evaluating `EpochNo` to WHNF should be enough to fully evaluate it.

Reply:
@KtorZ wrote:

> So this is the source of your question about NFData :)

Yes!

>  Looking more closely at the source code, OddWord is actually a newtype constructor around a Word32. Which promises no particular runtime overhead of the type itself. Meaning that evaluating EpochNo to WHNF should be enough to fully evaluate it.

Fair enough, I think this argument makes sense.